processing/src/main/java/org/apache/druid/query/scan/ScanQueryEngine.java

```diff
@@ -131,8 +131,8 @@
                     intervals.get(0),
                     query.getVirtualColumns(),
                     Granularities.ALL,
-                    query.getOrder().equals(ScanQuery.Order.DESCENDING) ||
-                    (query.getOrder().equals(ScanQuery.Order.NONE) && query.isDescending()),
+                    query.getTimeOrder().equals(ScanQuery.Order.DESCENDING) ||
+                    (query.getTimeOrder().equals(ScanQuery.Order.NONE) && query.isDescending()),
```

Comment:
I know this isn't new, but unless i'm missing something it looks like the scan query constructor hard codes `descending` to false, so the 2nd clause should never be true?

Reply:
I think you're right, which is good, because thinking about orderBy + a separate "descending" flag is definitely weird. I think it'd make sense to remove this branch when the logic is updated to be orderBy-focused in a later patch.